packages/custodyKeyring/src/CustodyKeyring.ts

```diff
@@ -109,21 +110,37 @@ export abstract class CustodyKeyring extends EventEmitter {
       this.selectedAddresses = migratedOpts.selectedAddresses || [];
       this.accountsDetails = migratedOpts.accountsDetails || [];
       this.meta = migratedOpts.meta || {};
+
+      // BEGIN UPDATE ACCOUNTS WITH ENV NAME
+      const custodians = this.getCustodians();
+      this.accountsDetails
+        .filter(account => !account.envName)
+        .forEach(account => {
+          account.envName = custodians.find(c => c.apiUrl === account.apiUrl)?.envName;
+        });
+
+      this.selectedAddresses
+        .filter(account => !account.envName)
+        .forEach(account => {
+          account.envName = custodians.find(c => c.apiUrl === account.apiUrl)?.envName;
+        });
+      // ENV UPDATE ACCOUNTS WITH ENV NAME
```

Comment:
I have tried doing this update as a migration, and it is indeed possible, however:
- Migrations are currently not working as they are, I had to do some changes to the `migrate` method and inject the list of custodians.
- Even after doing those changes, it felt flaky due to the fact that there are no currently working migrations and types are not well defined.
- IMO, it'd be best to tackle the migration system as its own PR, with a robust migration solution instead of an ad-hoc one.
- The way the update is done currently is safe and clear. Keyring data will update `envName` until that value is populated.